### DISCUSSION TITLE
How to use MVP for time series imputation?

### DISCUSSION THREAD

**Discussion by Roxyi** [timeseriesAI/tsai#459] [Q&A] [answered]
Hi, how can I use [MVP](https://github.com/timeseriesAI/tsai/blob/main/tutorial_nbs/08_Self_Supervised_TSBERT.ipynb) if the downstream task is imputing a time series?

**Comment by vrodriguezf** [comment]
In this case the pretrain task would be the same as the downstream task, so you can just use the learner from MVP to make inference (imputation) on new data.

**Comment by oguiza** [comment, marked as answer]
Hi @Roxyi,
Thanks for your question.
I've realized there's no example on how to impute data using MVP. 
I've created a simple [example](https://colab.research.google.com/gist/oguiza/5a9afbbe950e784423f9e98e838c7708/imputation_example.ipynb) to demonstrate how this could be done. I may add it to the tutorial nbs. 
It's simple, but there's are a few things to take into account. 
Please, let me know if it works for you, and if you have any additional questions.

**Reply by Roxyi** [reply]
Thank you, got it!

**Reply by Roxyi** [reply]
Hi @oguiza ,
Thank you for the example. The code works fine with no error but the predictions are not good. It seems that the predicted values of the masked positions are all very close to 0 while the neighboring points are far away from 0. I also compared the predicted values with the true values. As is shown in the following figures, the blue lines are the predicted values and the orange lines are the true values.
![image](https://user-images.githubusercontent.com/8086358/162681344-313f1281-ec12-487d-a608-1aacd9481a28.png)
![image](https://user-images.githubusercontent.com/8086358/162681377-a9b52a5c-f303-4ed9-b75f-7b1f50f986fe.png)
I tried to increase the number of epochs, but it didn't work. I tried other architectures like TSTPlus, it didn't work either. Is there any other thing that I can do to get a better result?
